src/controllers/hsm/hsm.controller.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+const {
+  Connection,
+  HathorWallet
+} = require('@hathor/wallet-lib');
+const {
+  initializedWallets,
+  hardWalletIds
+} = require('../../services/wallets.service');
+const { API_ERROR_CODES } = require('../../helpers/constants');
+const hsmService = require('../../services/hsm.service');
+const { HsmError } = require('../../errors');
+const settings = require('../../settings');
+const { notificationBus } = require('../../services/notification.service');
+
+/**
+ * Starts a read-only wallet integrated with the HSM.
+ * The process is similar to the one on the `/start` endpoint.
+ * @see src/controllers/index.controller.js , start function
+ */
+async function startHsmWallet(req, res) {
+  // Retrieving parameters from request body
+  const walletId = req.body['wallet-id'];
+  const hsmKeyName = req.body['hsm-key'];
+
+  // Validates input wallet-id
+  if (!walletId) {
+    res.send({
+      success: false,
+      message: 'Parameter \'wallet-id\' is required.',
+    });
+    return;
+  }
+  if (initializedWallets.has(walletId)) {
+    // We already have a wallet for this key
+    // so we log that it won't start a new one because
+    // it must first stop the old wallet and then start the new
+    console.error('Error starting wallet because this wallet-id is already in use. You must stop the wallet first.');
+    res.send({
+      success: false,
+      message: `Failed to start wallet with wallet id ${walletId}`,
+      errorCode: API_ERROR_CODES.WALLET_ALREADY_STARTED,
+    });
+    return;
+  }
+
+  // Validates input hsm-key
+  if (!hsmKeyName) {
+    res.send({
+      success: false,
+      message: 'Parameter \'hsm-key\' is required.',
+    });
+    return;
+  }
+
+  // Connects to the HSM
+  let connectionObj;
+  try {
+    connectionObj = await hsmService.hsmConnect();
+  } catch (e) {
+    const errorMessage = e instanceof HsmError
+      ? e.message
+      : `Unexpected error on HSM connection: ${e.message}`;
+
+    res.send({
+      success: false,
+      message: errorMessage,
+    });
+    return;
+  }
+
+  // Validates if the requested key is a valid BIP32 xPriv on the HSM
+  const validationObj = await hsmService.isKeyValidXpriv(connectionObj, hsmKeyName);
+  if (!validationObj.isValidXpriv) {
+    res.send({
+      success: false,
+      message: validationObj.reason,
+    });
+    return;
+  }
+
+  // Obtains this wallet's xPub from the HSM
+  const xPub = await hsmService.getXPubFromKey(connectionObj, hsmKeyName, {
+    isReadOnlyWallet: true,
+  });
+  await hsmService.hsmDisconnect();
+
+  // Builds the wallet configuration object
+  const config = settings.getConfig();
+  const walletConfig = {
+    xpub: xPub,
+    password: '123',
+    pinCode: '123',
+    multisig: null,
+    connection: null,
+  };
```

Comment:
suggestion: use the `getReadonlyWalletConfig` helper here.

I know it's not necessary, but since this helper is used by the normal start method it will be kept updated.

Reply:
Always good to reuse existing helper methods. Done on [519970f](https://github.com/HathorNetwork/hathor-wallet-headless/pull/356/commits/519970fb671308993a0b28efd0cf9633440081c4).